magento/magento2
# Check if the product exists in the system before adding to comparing list.

Check if the product exists in the system before adding to comparing list.

### Description (*)
Check if the product exists in the system before adding to comparing list. If the product doesn't exist, then we simply do nothing.

### Fixed Issues (if relevant)

1. magento/magento2#23071: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails

### Manual testing scenarios (*)

1. Use this url http:://www.domain.com/catalog/product_compare/index/items/ab
2. Use this url http:://www.domain.com/catalog/product_compare/index/items/1234567890
Make sure product ID doesn't exist in the system.

### Questions or comments


### Contribution checklist (*)
